application/forms/Jobs/ScheduleForm.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/* Icinga Web 2 X.509 Module | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\X509\Forms\Jobs;
+
+use DateTime;
+use Exception;
+use Icinga\Application\Icinga;
+use Icinga\Authentication\Auth;
+use Icinga\Module\X509\Common\Database;
+use Icinga\Module\X509\Model\X509Schedule;
+use Icinga\Util\Json;
+use Icinga\Web\Notification;
+use ipl\Html\HtmlElement;
+use ipl\Validator\CallbackValidator;
+use ipl\Web\Compat\CompatForm;
+use ipl\Web\FormElement\ScheduleElement;
+use ipl\Web\Url;
+
+use function ipl\Stdlib\get_php_type;
+
+class ScheduleForm extends CompatForm
+{
+    use Database;
+
+    /** @var int */
+    protected $jobId;
+
+    /** @var X509Schedule */
+    protected $schedule;
+
+    /** @var ScheduleElement */
+    protected $scheduleElement;
+
+    public function __construct(X509Schedule $schedule = null)
+    {
+        $this->schedule = $schedule;
+        $this->scheduleElement = new ScheduleElement('schedule_element');
+        $this->scheduleElement->setIdProtector([Icinga::app()->getRequest(), 'protectId']);
+    }
+
+    protected function isUpdating(): bool
+    {
+        return Url::fromRequest()->getPath() === 'x509/job/update-schedule';
+    }
+
+    public function setJobId(int $jobId): self
+    {
+        $this->jobId = $jobId;
+
+        return $this;
+    }
+
+    /**
+     * Get multipart updates
+     *
+     * @return array
+     */
+    public function getPartUpdates(): array
+    {
+        return $this->scheduleElement->prepareMultipartUpdate($this->getRequest());
+    }
+
+    public function hasBeenSubmitted(): bool
+    {
+        return $this->hasBeenSent() && ($this->getPopulatedValue('submit') || $this->getPopulatedValue('btn_remove'));
```

Comment:
`getPressedSubmitElement()` returns null in case of autosubmits, I just used `getPopulatedValue()` here instead of checking whether the pressed button is non-null and whether it matches one of these names.  `JobsConfigForm`, on the other hand, doesn't have any autosubmit elements, which is why I used the `getPressedSubmitElement()` function there.